grid2op/Opponent/RandomLineOpponent.py

```diff
@@ -0,0 +1,123 @@
+
+        self._do_nothing = self.action_space.actions[0]
+        self._attacks = self.action_space.actions[1:]
+
+    def init(self, *args, **kwargs):
+        """
+        Generic function used to initialize the derived classes. For example, if an opponent reads from a file, the
+        path where is the file is located should be pass with this method.
+        """
+        pass
+
+    def reset(self, initial_budget):
+        """
+        This function is called at the end of an episode, when the episode is over. It aims at resetting the
+        self and prepare it for a new episode.
+
+        Parameters
+        ----------
+        initial_budget: ``float``
+            The initial budget the opponent has
+        """
+        pass
+
+    def attack(self, observation, agent_action, env_action, budget, previous_fails):
+        """
+        This method is the equivalent of "attack" for a regular agent.
+
+        Opponent, in this framework can have more information than a regular agent (in particular it can
+        view time step t+1), it has access to its current budget etc.
+
+        Parameters
+        ----------
+        observation: :class:`grid2op.Observation.Observation`
+            The last observation (at time t)
+
+        opp_reward: ``float``
+            THe opponent "reward" (equivalent to the agent reward, but for the opponent) TODO do i add it back ???
+
+        done: ``bool``
+            Whether the game ended or not TODO do i add it back ???
+
+        agent_action: :class:`grid2op.Action.Action`
+            The action that the agent took
+
+        env_action: :class:`grid2op.Action.Action`
+            The modification that the environment will take.
+
+        budget: ``float``
+            The current remaining budget (if an action is above this budget, it will be replaced by a do nothing.
+
+        previous_fails: ``bool``
+            Wheter the previous attack failed (due to budget or ambiguous action)
+
+        Returns
+        -------
+        attack: :class:`grid2op.Action.Action`
+            The attack performed by the opponent. In this case, a do nothing, all the time.
+        """
+        # TODO maybe have a class "GymOpponent" where the observation would include the budget  and all other
+        # TODO information, and forward something to the "act" method.
+
+        if observation is None: # during creation of the environment
+            return self._do_nothing # do nothing
+
+        action_line_ids = [a.as_dict()['set_line_status']['disconnected_id'][0]
+                           for a in self._attacks]
+        status = observation.line_status[action_line_ids]
+
+        # If all lines are disconnected
+        if not any(status):
+            return self._do_nothing
+
+        # Pick a line among the connected lines
+        return np.random.choice(self._attacks)
+
+    def tell_attack_continues(self, observation, agent_action, env_action, budget):
```

Comment:
no need to overload it if it does nothing more than the original class (in this case nothing ;-) )